packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/ExtractionService.java

```diff
@@ -498,84 +499,96 @@ public Future<Model> alignAMR(
+					modelService.getAsset(modelId, hasWritePermission).orElseThrow();
 
-				final ResponseEntity<JsonNode> res;
-				try {
-					res = skemaUnifiedProxy.linkAMRFile(amrFile, extractionFile);
-				} catch (final FeignException e) {
-					final String error = "Transitive service failure";
-					log.error(error, e.contentUTF8(), e);
-					throw new ResponseStatusException(
-							e.status() < 100 ? HttpStatus.INTERNAL_SERVER_ERROR : HttpStatus.valueOf(e.status()),
-							error + ": " + e.getMessage());
-				}
-				if (!res.getStatusCode().is2xxSuccessful()) {
-					throw new ResponseStatusException(res.getStatusCode(), "Unable to link AMR file");
-				}
+			final String modelString = objectMapper.writeValueAsString(model);
 
-				final JsonNode modelJson = objectMapper.valueToTree(model);
+			if (document.getMetadata() == null) {
+				document.setMetadata(new HashMap<>());
+			}
+			if (document.getMetadata().get("attributes") == null) {
+				throw new RuntimeException("No attributes found in document");
+			}
 
-				// overwrite all updated fields
-				JsonUtil.recursiveSetAll((ObjectNode) modelJson, res.getBody());
+			final JsonNode attributes =
+					objectMapper.valueToTree(document.getMetadata().get("attributes"));
 
-				// update the model
-				modelService.updateAsset(model, hasWritePermission);
+			final ObjectNode extractions = objectMapper.createObjectNode();
+			extractions.set("attributes", attributes);
 
-				// create provenance
-				final Provenance provenance = new Provenance(
-						ProvenanceRelationType.EXTRACTED_FROM,
-						modelId,
-						ProvenanceType.MODEL,
-						documentId,
-						ProvenanceType.DOCUMENT);
-				provenanceService.createProvenance(provenance);
+			final String extractionsString = objectMapper.writeValueAsString(extractions);
 
-				return model;
+			final StringMultipartFile amrFile = new StringMultipartFile(modelString, "amr.json", "application/json");
+			final StringMultipartFile extractionFile =
+					new StringMultipartFile(extractionsString, "extractions.json", "application/json");
 
+			final ResponseEntity<JsonNode> res;
+			try {
+				res = skemaUnifiedProxy.linkAMRFile(amrFile, extractionFile);
 			} catch (final FeignException e) {
 				final String error = "Transitive service failure";
 				log.error(error, e.contentUTF8(), e);
 				throw new ResponseStatusException(
 						e.status() < 100 ? HttpStatus.INTERNAL_SERVER_ERROR : HttpStatus.valueOf(e.status()),
 						error + ": " + e.getMessage());
-			} catch (final Exception e) {
-				final String error = "SKEMA link_amr request from document: " + documentId + " failed.";
-				log.error(error, e);
-				notificationInterface.sendError(error + " — " + e.getMessage());
-				throw new ResponseStatusException(org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, error);
 			}
-		});
+			if (!res.getStatusCode().is2xxSuccessful()) {
+				throw new ResponseStatusException(res.getStatusCode(), "Unable to link AMR file");
```

Comment:
In the past some of the TA3 services have returned -1 as an HTTP error code. I see there is safety for this above, but, we may want it here too.